rustfst/src/fst_traits/allocable_fst.rs

```diff
@@ -0,0 +1,39 @@
+use failure::Fallible;
+use crate::StateId;
+
+/// Trait defining the methods to control allocation for a wFST
+pub trait AllocableFst {
```

Comment:
Don't we want to add a trait bound on the `Fst` trait ?

Reply:
`StateIterator` is not bounded too and `ArcIterator` is only on CoreFst. We could bound all of them to the `Fst` trait ?

Reply:
CoreFst exists only to avoid a cyclic dependency on the trait bound. 
Fst -> StateIterator -> CoreFst
Fst -> ArcIterator -> CoreFst
AllocableFst bounded on the Fst trait seems fine

Reply:
I have added the bounding to the AllocableFst